sequencer.c

```diff
@@ -2653,6 +2653,26 @@ int sequencer_rollback(struct repository *r, struct replay_opts *opts)
 	return -1;
 }
 
+int sequencer_skip(struct repository *r, struct replay_opts *opts)
+{
+	struct child_process reset = CHILD_PROCESS_INIT;
```

Comment:
@tgummerer @newren 
Can I replace spawning "git reset --merge" here with calling `rollback_single_pick()`?

Reply:
It looks like `rollback_single_pick()` does indeed pretty much do that, with some additional check.  Before further refactoring I'd write some tests showing how this works though, and then doing the refactoring.  At that point you should be able to iterate faster, and be fairly confident that what you're doing is actually the right thing :)

Reply:
Yes, currently I am writing and improving tests. I have named the test file as "t3514-revert-cherry-pick-skip.sh", is it right?

Reply:
I'd probably put it in the existing `t3510`, which seems to fit this use case quite well.

Reply:
Sorry, I didn't see it earlier. Since `git rebase --skip` has its own test file, so, I thought, why not give `git [cherry-pick | revert] --skip` its own separate test file? Anyway, I have committed, have a look and suggest further changes. Thanks!